src/other/index.jl

```diff
@@ -214,8 +214,28 @@ end
 @inline Base.getindex(x::AbstractIndex, idx::AbstractRange{<:Integer}) =
     getindex(x, collect(Int, idx))
 @inline Base.getindex(x::AbstractIndex, ::Colon) = Base.OneTo(length(x))
-@inline Base.getindex(x::AbstractIndex, notidx::Not) =
+
+@inline Base.getindex(x::AbstractIndex, notidx::Not{<:AbstractVector}) =
+    setdiff(1:length(x), [getindex(x, idx) for idx in notidx.skip if haskey(x, idx)])
+@inline function Base.getindex(x::AbstractIndex, notidx::Not{<:All})
+    if isempty(notidx.skip.cols) 
+        return Int[]
+    else 
+        intersect(getindex.(Ref(x), Not.(notidx.skip.cols))...)
+    end
+end
+@inline Base.getindex(x::AbstractIndex, notidx::Not{<:Between}) =
+    return setdiff(1:length(x), getindex(x, notidx.skip))
+@inline Base.getindex(x::AbstractIndex, notidx::Not{<:Regex}) =
     setdiff(1:length(x), getindex(x, notidx.skip))
+@inline function Base.getindex(x::AbstractIndex, notidx)
```

Comment:
I guess you meant this:
```suggestion
@inline function Base.getindex(x::AbstractIndex, notidx::Union{Not{Symbol},Not{<:AbstractString}})
```